App.js

```diff
@@ -68,6 +70,45 @@ const location = {
 };
 
 function HomeScreen({ navigation }) {
+  const [geoLocation, setGeoLocation] = useState(null);
+  const [errorLocationMsg, setErrorLocationMsg] = useState(null);
+
+  function convertToMapLocation(geoLoc) {
+    if (geoLoc) {
+      console.log(geoLoc);
+      return {
+        latitude: geoLoc.coords.latitude,
+        longitude: geoLoc.coords.longitude,
+        latitudeDelta: 0.009,
+        longitudeDelta: 0.009,
+      };
+    }
+  }
+
+  useEffect(() => {
+    (async () => {
+      let { status } = await Location.requestForegroundPermissionsAsync();
+      if (status !== "granted") {
+        setErrorLocationMsg("Permission to access location was denied");
+        return;
+      }
+
+      let newLocation = await Location.getCurrentPositionAsync({});
+      setGeoLocation(convertToMapLocation(newLocation));
+    })();
+  }, []);
+
+  let myLocation = null;
+  if (errorLocationMsg) {
+    alert(errorLocationMsg);
+  } else if (geoLocation) {
+    myLocation = {
+      latitude: geoLocation.latitude,
+      longitude: geoLocation.longitude,
+      latitudeDelta: 0.009,
+      longitudeDelta: 0.009,
+    };
```

Comment:
Isn't this the same code in the convert function that makes the geolocation variable that's being used in the map component?

It looks like you're halfway done refactoring (and you skipped the best part - deleting code!)